src/core/components/Dropzone/Dropzone.tsx

```diff
@@ -36,29 +41,29 @@ const Dropzone = (props: DropzoneProps) => {
     onChange,
   } = props;
   const { t } = useTranslation();
-  const [isMounted, setMounted] = useState(false);
-  const [inputKey, setInputKey] = useState("");
-  const { getInputProps, getRootProps, acceptedFiles, fileRejections } =
+  const [inputKey] = useState("");
+  const [acceptedFilesMap, setAcceptedFilesMap] = useState<
+    Map<string, FileWithPath>
+  >(new Map());
+  const acceptedFiles = acceptedFilesMap.values().toArray();
+  const { getInputProps, getRootProps, fileRejections, isDragAccept } =
     useDropzone({
       validator,
       accept,
       maxFiles,
       disabled,
       useFsAccessApi: false,
       multiple: maxFiles !== 1,
+      onDrop: (newAcceptedFiles, fileRejections) => {
+        const newAcceptedFilesMap = new Map(acceptedFilesMap);
+        newAcceptedFiles.forEach(
+          (file) => newAcceptedFilesMap.set(file.name, file), //  Ignore duplicates by names
+        );
+        setAcceptedFilesMap(newAcceptedFilesMap);
+        onChange(newAcceptedFilesMap.values().toArray(), fileRejections);
+      },
     });
 
-  useEffect(() => {
-    setMounted(true);
-  }, []);
-
-  useEffect(() => {
-    if (isMounted) {
-      onChange(acceptedFiles, fileRejections);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [acceptedFiles, fileRejections]);
-
```

Comment:
Good to reduce usage of `useEffect` when we can. Here we can just chain the `onChange` on the `onDrop`

Reply:
I do not even remember why it was implemented. If only "old me" had written a comment...

Reply:
Since the callback of useDropzone was not used, I think that it is the only pattern in React to perform this